### PR TITLE
cmd: Add flags for tx and receipt pruning

### PR DESCRIPTION
**Pull request by 2dvorak**
## Proposed changes

- This PR adds flags for enabling tx/receipt pruning and setting their pruning retention
- Actual logics for the pruning will be added in the follow-up PR

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/kaiachain/kaia/blob/main/CONTRIBUTING.md) doc
- [x] I have read the [CLA](https://gist.github.com/kaiachain-dev/bbf65cc330275c057463c4c94ce787a6) and signed by comment `I have read the CLA Document and I hereby sign the CLA` in first time contribute
- [ ] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

none

### CODE REVIEW

File: cmd/utils/config.go [hunk 1]
```diff
@@ -604,6 +604,18 @@ func (kCfg *KaiaConfig) SetKaiaConfig(ctx *cli.Context, stack *node.Node) {
...
+		cfg.TxPruningRetention = ctx.Uint64(TxPruningRetentionFlag.Name)
+	} else {
+		cfg.TxPruningRetention = 0
+	}
+	cfg.ReceiptPruning = ctx.Bool(ReceiptPruningFlag.Name)
+	if cfg.ReceiptPruning {
+		cfg.ReceiptPruningRetention = ctx.Uint64(ReceiptPruningRetentionFlag.Name)
+	} else {
+		cfg.ReceiptPruningRetention = 0
+	}
```
Comment (hyunsooda) [inline cmd/utils/config.go:618]: [About value constraint]

- Even if both retention values differ, I guess no issue will be found. Do you think another point of view about it?
- Is it okay if both values are lower than the `LivePruningRetention` value?
- A plan of upper and lower bound?

Comment (blukat29) [reply cmd/utils/config.go:618]: - I think they can differ. At least I know an extreme case where deleting all receipts and block syncs well. In pruning terms, TxPruningRetention = infinity and ReceiptPruningRetention = 0.
- Maybe? The Retention value only affects the [pruneTrieNodeLoop](https://github.com/kaiachain/kaia/blob/35e43b859b658732323d3f67264c46a1475016a0/blockchain/blockchain.go#L1509) and it is unrelated to txs or receipts in the database.
- No need for the upper bound. You can set 9999999999 to signify infinity. However, a lower bound could be a good safety measure. e.g. any arbitrary number like 128.

@2dvorak Can we have the lower bound for all 3 (tx, receipt, state live pruning) retentions?

Comment (blukat29) [reply cmd/utils/config.go:618]: @2dvorak @hyunsooda  I think separate lower-bound PR is better. So I can mention it in the Release note.

**Comment by blukat29** [comment]
@2dvorak Rebase to the latest dev to fix linter.

File: cmd/utils/config.go (hunk 1 above)
Comment (2dvorak) [reply cmd/utils/config.go:618]: @hyunsooda @blukat29 OK, I'll create another PR for lower-bounds. Thanks!

**Review by blukat29 — APPROVED** [review]

**Review by ian0371 — APPROVED** [review]

File: cmd/utils/config.go (hunk 1 above)
Comment (blukat29) [reply cmd/utils/config.go:618]: @hyunsooda PTAL

**Review by hyeonLewis — APPROVED** [review]

**Review by hyunsooda — APPROVED** [review]

**Pull request merged**